storage/pebble/register.go

```diff
@@ -0,0 +1,126 @@
+package pebble
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/onflow/atree"
+
+	"github.com/onflow/flow-go/fvm/evm/types"
+)
+
+var _ types.BackendStorage = &Register{}
+
+type Register struct {
+	store  *Storage
+	height uint64
+	batch  *pebble.Batch
+}
+
+// NewRegister creates a new index instance at the provided height, all reads and
+// writes of the registers will happen at that height.
+// this is not concurrency safe.
+func NewRegister(
+	store *Storage,
+	height uint64,
+	batch *pebble.Batch,
+) *Register {
+	return &Register{
+		store:  store,
+		height: height,
+		batch:  batch,
+	}
+}
+
+func (r *Register) GetValue(owner, key []byte) ([]byte, error) {
+	var db pebble.Reader = r.store.db
+	if r.batch != nil {
+		db = r.batch
+	}
+
+	iter, err := db.NewIter(&pebble.IterOptions{
+		LowerBound: r.idLower(owner, key),
+		UpperBound: r.idUpper(owner, key),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create register range iterator: %w", err)
+	}
+	defer func() {
+		if err := iter.Close(); err != nil {
+			r.store.log.Error().Err(err).Msg("failed to close register iterator")
+		}
+	}()
+
+	found := iter.Last()
```

Comment:
This is interesting. We are using a different approach than[ how EN indexes the registers. ](https://github.com/onflow/flow-go/blob/master/storage/pebble/registers.go#L90)


EN uses `SeekPrefixGE`, where as here we use `iterator.Last`, 

I don't know the performance of iter.Last. I'm curious to see a benchmark of the two approach, see if they are the same. 

And what's your opinion of reusing the same logic from flow-go register store? Do we want to keep the same database scheme or a different one?

Reply:
Thanks! I checked the logic in flow-go and mostly copied it.

I copied it instead of reusing it, because:
- I don't need the owner field
- I don't need the code field
- I need different batch handling
- I don't need pruning yet

Please check again.